src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs

```diff
@@ -416,11 +416,6 @@ public static async Task<IEnumerable<SymbolCallerInfo>> FindCallersAsync(ISymbol
                 }
             }
 
-            if (searchSymbol.Kind == SymbolKind.NamedType && symbolToMatch.IsConstructor())
```

Comment:
📝 I tracked this back to the start of the repository:

https://github.com/dotnet/roslyn/blob/fdcbb654875a4c61312a4d72da8fb59fc15fd5a9/src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs#L315-L318

However, it seems to duplicate the intent of `NamedTypeSymbolReferenceFinder.DetermineCascadedSymbolsAsync`. Cascaded symbols are correctly aggregated by the target of the reference, thus addressing the original bug without locating duplicate references.

Reply:
That's great to know.  Mostly, i'm fine if we land in an area where:

1. the actual FAR **API** is sensible and clear.
2. the FAR feature/UI can handle these cases sensibly.  

in terms of the impl to get there, i'm very open to discussing options both in the FAR API model, as well as the code that runs at all teh different layers.  

I appreciate you looking at this, especially if we take a 'lean back and look at the whole picture' to decide what the best way forward is.